onnx/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/onnx/OnnxInferenceModel.kt

```diff
@@ -127,21 +152,27 @@ public open class OnnxInferenceModel : InferenceModel() {
      * @return Vector that represents the probability distributions of a list of potential outcomes
      */
     public fun predictSoftly(inputData: FloatArray): FloatArray {
-        require(::inputShape.isInitialized) { "Reshape functions is missed! Define and set up the reshape function to transform initial data to the model input." }
+        return predictSoftly(inputData, session.outputNames.first())
+    }
 
+    private fun predictSoftly(inputData: FloatArray, outputTensorIdx: Int): FloatArray {
```

Comment:
I think it would be a good idea to unify this method with `predictRaw` (as a separate PR).

Reply:
Agree, but the 'predictRaw' method currently has lower performance than other methods. So it requires some research to refactor it properly.